systemd/systemd.go

```diff
@@ -593,33 +597,82 @@ func (s *systemd) getGlobalUserStatus(unitNames ...string) ([]*UnitStatus, error
 	return sts, nil
 }
 
-func (s *systemd) CurrentMemoryUsage(unit string) (quantity.Size, error) {
-	out, err := s.systemctl("show", "--property", "MemoryCurrent", unit)
+func (s *systemd) getPropertyStringValue(unit, key string) (string, error) {
+	// XXX: ignore stderr of systemctl command to avoid further infractions
+	//      around LP #1885597
+	out, err := s.systemctl("show", "--property", key, unit)
 	if err != nil {
-		return 0, osutil.OutputErr(out, err)
+		return "", osutil.OutputErr(out, err)
 	}
-	// strip the MemoryCurrent= from the output
-	splitVal := strings.SplitN(strings.TrimSpace(string(out)), "=", 2)
+	cleanVal := strings.TrimSpace(string(out))
+
+	// strip the TasksCurrent= from the output
+	splitVal := strings.SplitN(cleanVal, "=", 2)
 	if len(splitVal) != 2 {
-		return 0, fmt.Errorf("invalid property format from systemd for MemoryCurrent")
+		return "", fmt.Errorf("invalid property format from systemd for %s (got %s)", key, cleanVal)
+	}
+
+	return strings.TrimSpace(splitVal[1]), nil
+}
+
+func (s *systemd) CurrentTasksCount(unit string) (int, error) {
+	tasksStr, err := s.getPropertyStringValue(unit, "TasksCurrent")
+	if err != nil {
+		return 0, err
 	}
 
-	trimmedVal := strings.TrimSpace(splitVal[1])
+	// if the unit is inactive or doesn't exist, the tasks current can be
+	// reported as "[not set]"
+	if tasksStr == "[not set]" {
```

Comment:
Given that this [not set] special casing is also used in CurrentMemoryUsage I wonder if there should be a specialized `getPropertyIntValue(unit, propertyName)` helper?

Reply:
Fine for a followup too of course fwiw, especially since this one is green :)

Reply:
yeah it's a good idea, I'll take care of it in a followup thanks for the suggestion

Reply:
this seems indeed a good idea for a follow up

Reply:
done in https://github.com/snapcore/snapd/pull/10379